test/test.do

```diff
@@ -1,418 +1,421 @@
+
+* A common axis plot with labels
+gen y2 = y + 5
+xtevent y eta, policyvar(z) timevar(t) window(5)
+loc lab : di %-9.2f `=e(y1)'
+loc lab=strtrim("`lab'")
+xteventplot, nominus1label ylab(-3 0 `"0 (`lab')"' 3) name(g1) /* " */
+xtevent y2 eta, policyvar(z) timevar(t) window(5)
+loc lab : di %-9.2f `=e(y1)'
+loc lab=strtrim("`lab'")
+xteventplot, nominus1label ylab(-3 0 `"0 (`lab')"' 3) name(g2) /* " */
+graph combine g1 g2
+drop y2
+
+* Test if/in
+xtevent y eta if i<100, panelvar(i) timevar(t) policyvar(z) window(5) plot
+xtevent y eta in 1/600 , panelvar(i) timevar(t) policyvar(z) window(5) plot
+xtevent y eta in 1/600 if i<30 , panelvar(i) timevar(t) policyvar(z) window(5) plot
+
+* Test nofe, note
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) nofe plot
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) nofe note plot
+
+* Test smoothest line
+xtevent eta, panelvar(i) timevar(t) policyvar(z) window(5) 
+xteventplot, smpath(scatter)
+xteventplot, smpath(line)
+xteventplot, smpath(line, technique("nr 10 bfgs 10"))
+
+* Test more suptreps
+
+cap graph drop g1
+cap graph drop g2
+
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(3) 
+xteventplot, suptreps(20) name(g1)
+xteventplot, suptreps(1e6) name(g2)
+
+graph combine g1 g2, rows(1)
+
+graph drop g1
+graph drop g2
+
+* Test savek
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) savek(a)
+des a_eq*, s
+des a_evtime, s
+drop a*
+
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) savek(b)
+
+* Test factor variables in varlist
+cap gen pois=rpoisson(5)
+xtevent y eta i.pois, panelvar(i) timevar(t) policyvar(z) window(5) plot
+cap drop pois
+* Test time series variables in varlist
+
+xtevent y l.eta , panelvar(i) timevar(t) policyvar(z) window(5) plot
+xtevent y f.eta , panelvar(i) timevar(t) policyvar(z) window(5) plot
+
+* Test asymmetric window
+xtevent y eta , panelvar(i) timevar(t) policyvar(z) window(-4 2) plot
+
+* Test normalizations
+
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) norm(-1) plot
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) norm(-2) plot
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) norm(-6) plot
+* xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) norm(-7) plot
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) norm(1) plot 
+xtevent y eta, panelvar(i) timevar(t) policyvar(z) window(5) norm(5)plot 
+
+graph drop _all
+
+* Test exclusion of unbalanced units with ambiguous eventtime
+gen z2 = z
+replace z2 = . if i==1 & t==7 
+xtevent y eta, policyvar(z2) window(5) 
+drop z2
+
+* Test estimate witthout staggered adoption
+xtevent y eta , panelvar(i) timevar(t) policyvar(z) window(3) nostaggered plot
+
+* Test reghdfe
+xtevent y eta , panelvar(i) timevar(t) policyvar(z) window(3) reghdfe plot
+
+* Test reghdfe and proxy and absorbing a variable 
+xtevent y eta, policyvar(z) window(3) proxy(x) nofe addabsorb(i) reghdfe

```

Comment:
@Constantino-Carreto-Romero can we change the variable added in -addabsorb- to another variable different from i? If we use i, we are actually not adding additional fixed effects to be absorbed.